### PR TITLE
make extend, combineJSON and parsers assync

### PR DESCRIPTION
**Pull request by jorenbroekema**

### CODE REVIEW

File: lib/utils/combineJSON.js [hunk 1]
```diff
@@ -60,14 +60,19 @@ function combineJSON(arr, deep, collision, source, parsers=[]) {
...
-        if (resolvedPath.match(pattern)) {
-          file_content = parse({
+      // Allows async parsers
+      for (let i = 0; i < parsers.length; i++) {
+        if (resolvedPath.match(parsers[i].pattern)) {
+          let result = parsers[i].parse({
             contents: fs.readFileSync(resolvedPath, {encoding:'UTF-8'}),
             filePath: resolvedPath
           });
+          if (result instanceof Promise) {
```
Comment (gqio) [inline lib/utils/combineJSON.js]: you can await even the result is not a promise IIRC. simplifying this call

File: lib/utils/combineJSON.js [hunk 2]
```diff
@@ -60,14 +60,19 @@ function combineJSON(arr, deep, collision, source, parsers=[]) {
 
       // Iterate over custom parsers, if the file path matches the parser's
       // pattern regex, use it's parse function to generate the object
-      parsers.forEach(({pattern, parse}) => {
-        if (resolvedPath.match(pattern)) {
-          file_content = parse({
+      // Allows async parsers
+      for (let i = 0; i < parsers.length; i++) {
```
Comment (gqio) [inline lib/utils/combineJSON.js]: I think 
```
for(const parser of parsers.filter(p=>resolvePath.match(p.pattern))){
  file_content = await parser.parse({
    contents: fs.readFileSync(resolvedPath, {encoding:'UTF-8'}),
    filePath: resolvedPath
  });
}
```
 is more readable IMHO

Comment (jorenbroekema) [reply lib/utils/combineJSON.js]: `for of` I agree, one-lining the conditional into it, debatable, but I'll take it :)

File: lib/utils/combineJSON.js (hunk 1 above)
Comment (jorenbroekema) [reply lib/utils/combineJSON.js]: TIL lol, you're right.

**Review by gqio — APPROVED** [review]

**Pull request merged**